[openmvg] Build error M1 MacOS

**Host Environment**

- OS: macOS (version)
- Chip: Apple M1

**To Reproduce**

Steps to reproduce the behavior:

1. Run `./vcpkg install openmvs --allow-unsupported` command in the terminal.

**Failure logs**


```
./vcpkg install openmvs --allow-unsupported

Computing installation plan...
warning: openmvg is only supported on '!(osx & arm64)', which does not match arm64-osx. This usually means that there are known build failures, or runtime problems, when building other platforms. Proceeding anyway due to `--allow-unsupported`.
The following packages will be built and installed:
  * openmvg[core,software]:arm64-osx -> 2.0#6
    openmvs[core,nonfree]:arm64-osx -> 2.1.0
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet arm64-osx...
Restored 0 package(s) from /Users/yusuf/.cache/vcpkg/archives in 8.08 us. Use --debug to see more details.
Installing 1/2 openmvg:arm64-osx...
Building openmvg[core,software]:arm64-osx...
warning: -- Using community triplet arm64-osx. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: /Users/yusuf/vcpkg/triplets/community/arm64-osx.cmake
-- Using cached openMVG-openMVG-d0fe73dd426ae4001631a51272cff71047522df9.tar.gz.
-- Cleaning sources at /Users/yusuf/vcpkg/buildtrees/openmvg/src/1047522df9-5f641eb7aa.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /Users/yusuf/vcpkg/downloads/openMVG-openMVG-d0fe73dd426ae4001631a51272cff71047522df9.tar.gz
-- Applying patch build_fixes.patch
-- Applying patch 0001-eigen_3.4.0.patch
-- Applying patch 0002-eigen-3.4.patch
-- Applying patch no-absolute-paths.patch
-- Applying patch fix-coinutils.patch
-- Using source at /Users/yusuf/vcpkg/buildtrees/openmvg/src/1047522df9-5f641eb7aa.clean
-- Configuring arm64-osx
-- Building arm64-osx-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: /opt/homebrew/Cellar/cmake/3.26.2/bin/cmake --build . --config Debug --target install -- -v -j9
    Working Directory: /Users/yusuf/vcpkg/buildtrees/openmvg/arm64-osx-dbg
    See logs for more information:
      /Users/yusuf/vcpkg/buildtrees/openmvg/install-arm64-osx-dbg-out.log

Call Stack (most recent call first):
  installed/arm64-osx/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/arm64-osx/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/openmvg/portfile.cmake:79 (vcpkg_cmake_install)
  scripts/ports.cmake:147 (include)


error: building openmvg:arm64-osx failed with: BUILD_FAILED
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+openmvg
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?title=[openmvg]+Build+error&body=Copy+issue+body+from+%2FUsers%2Fyusuf%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md

```

# log

https://codeshare.io/Pdxe1e

```
/Library/Developer/CommandLineTools/usr/lib/clang/14.0.0/include/immintrin.h:14:2: error: "This header is only meant to be used on x86 and x64 architecture"
#error "This header is only meant to be used on x86 and x64 architecture"
```

--- a/ports/openmvg/build_fixes.patch
+++ b/ports/openmvg/build_fixes.patch
@@ -639,3 +639,18 @@ index 8a69c2344..1ed9a21fe 100644
          FIND_LIBRARY(OSICLP_LIBRARY NAMES OsiClp)
  
          # locate Clp libraries
+
+diff --git a/src/openMVG/matching/metric_simd.hpp b/src/openMVG/matching/metric_simd.hpp
+index 7e09f6ed..916c84e0 100644
+--- a/src/openMVG/matching/metric_simd.hpp
++++ b/src/openMVG/matching/metric_simd.hpp
+@@ -17,7 +17,9 @@
+ #include <numeric>
+
+ #include <cstdint>
++#if defined(__amd64__) || defined(__i386__)
+ #include <immintrin.h>
++#endif
+
+ namespace openMVG {
+ namespace matching {

--- a/ports/openmvg/vcpkg.json
+++ b/ports/openmvg/vcpkg.json
@@ -1,10 +1,10 @@
 {
   "name": "openmvg",
   "version": "2.0",
-  "port-version": 8,
+  "port-version": 9,
   "description": "open Multiple View Geometry library. Basis for 3D computer vision and Structure from Motion.",
   "license": "MPL-2.0-no-copyleft-exception",
-  "supports": "(x86 | x64) & !xbox",
+  "supports": "(x86 | x64 | arm64) & !xbox",
   "dependencies": [
     "cereal",
     {

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -6046,7 +6046,7 @@
     },
     "openmvg": {
       "baseline": "2.0",
-      "port-version": 8
+      "port-version": 9
     },
     "openmvs": {
       "baseline": "2.1.0",

--- a/versions/o-/openmvg.json
+++ b/versions/o-/openmvg.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "6ddaa3bedcc57eb498c9e180bb8546a2726a1ddc",
+      "version": "2.0",
+      "port-version": 9
+    },
+    {
       "git-tree": "3712d684593f777107e58a37bf601083a6351738",
       "version": "2.0",
       "port-version": 8

